### PR TITLE
[MR-2086] State can specify state programs for rate cert

### PR DESCRIPTION
**Pull request by JasonLin0991**
## Summary
[MR-2086](https://qmacbis.atlassian.net/browse/MR-2086)

- State users can specify state programs that apply to their rate certification.
- Refactor program combobox/dropdown to `ProgramSelect` component.

#### Test cases covered

- Unit Tests
   - `RateDetials.test.tsx`
      - 'displays program options based on current user state'
   - `RateDetailsSummarySection.test.tsx`
      - 'renders programs that apply to rate certification'
   - `ProgramSelect.test.tsx`
      - 'displays program options'
      - 'can select and return programs in an array'
      - 'can remove all selected programs'
- Cypress tests
   - Update commands to select program on rate detail page
      - `fillOutNewRateCertification`
      - `fillOutAmendmentToPriorRateCertification`

### CODE REVIEW

File: services/app-web/src/components/ProgramSelect/ProgramSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+interface ProgramOption {
+    readonly value: string
+    readonly label: string
+    readonly isFixed?: boolean
+    readonly isDisabled?: boolean
+}
+
+export const ProgramSelect = ({
+    statePrograms,
```
Comment (haworku) [inline services/app-web/src/components/ProgramSelect/ProgramSelect.tsx:23]: This component's props should be more extendable to be re-useable across the application. We should be able to pass in more the HTML attributes (aria labels, custom ids, etc) and override the defaults-  see `FieldTextInput` as an example.

File: services/app-web/src/hooks/useStatePrograms.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { useAuth } from '../contexts/AuthContext'
+import { Program } from '../gen/gqlClient'
+
+// Determine current route name type (e.g. SUBMISSION_TYPE) using the getRouteName utility
```
Comment (haworku) [inline services/app-web/src/hooks/useStatePrograms.ts:4]: nit: update the commented description here

**Review by haworku — APPROVED** [review]

**Comment by haworku** [comment]
Please do not merge without determining if we need to add a feature flag to this epic to ship incrementally. I commented on that here https://qmacbis.atlassian.net/browse/MR-2086?focusedCommentId=112570

**Comment by JasonLin0991** [comment]
> Please do not merge without determining if we need to add a feature flag to this epic to ship incrementally. I commented on that here https://qmacbis.atlassian.net/browse/MR-2086?focusedCommentId=112570

You got it! I have the feature flag implemented, I'll push up once we if we determine that we do want it.

File: services/app-web/src/common-code/proto/healthPlanFormDataProto/unlockedHealthPlanFormDataSchema.ts [hunk 3]
```diff
@@ -167,4 +167,5 @@ export const unlockedHealthPlanFormDataSchema = z.object({
     rateDateEnd: z.date().optional(),
     rateDateCertified: z.date().optional(),
     rateAmendmentInfo: rateAmendmentInfoSchema.optional(),
+    rateProgramIDs: z.array(z.string()).optional(),
```
Comment (macrael) [inline services/app-web/src/common-code/proto/healthPlanFormDataProto/unlockedHealthPlanFormDataSchema.ts:170]: does this work if you drop the optional() here? I think we are going to decode this into an empty array every time so the array itself doesn't need to be optional.

File: services/app-web/src/hooks/useStatePrograms.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+// Get state programs from logged in state users data
+const useStatePrograms = (): Program[] | [] => {
+    const { loggedInUser } = useAuth()
+    let statePrograms: Program[] = []
+
+    if (loggedInUser && loggedInUser.__typename === 'StateUser') {
+        statePrograms = loggedInUser.state.programs
+    } else {
+        console.error(
```
Comment (macrael) [inline services/app-web/src/hooks/useStatePrograms.ts:12]: thanks for this. If it were more serious, coding errors are the kind of things I think it's appropriate to `throw` for.

File: services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.tsx [hunk 5]
```diff
@@ -402,6 +409,42 @@ export const RateDetails = ({
...
+                                    <Label htmlFor="programIDs">
+                                        Programs this rate certification covers
+                                    </Label>
+                                    {showFieldErrors(errors.rateProgramIDs) && (
+                                        <PoliteErrorMessage>
+                                            {errors.rateProgramIDs}
+                                        </PoliteErrorMessage>
+                                    )}
+                                    <Field name="programIDs">
+                                        {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
```
Comment (macrael) [inline services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.tsx]: have we used this syntax elsewhere? Wrapping a component in `<Field>` is new to me

**Review by macrael — APPROVED** [review]

File: services/app-web/src/common-code/proto/healthPlanFormDataProto/unlockedHealthPlanFormDataSchema.ts (hunk 3 above)
Comment (JasonLin0991) [reply services/app-web/src/common-code/proto/healthPlanFormDataProto/unlockedHealthPlanFormDataSchema.ts]: Looks like it does!

File: services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.tsx (hunk 5 above)
Comment (JasonLin0991) [reply services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.tsx]: This was used in SubmissionType, I replaced the `Select` component with our `ProgramSelect` essentially the same.

**Pull request merged**